base/range.jl

```diff
@@ -1038,3 +1038,27 @@ function +(r1::StepRangeLen{T,S}, r2::StepRangeLen{T,S}) where {T,S}
 end
 
 -(r1::StepRangeLen, r2::StepRangeLen) = +(r1, -r2)
+
+# Modular arithmetic on ranges
+
+"""
+    mod(x::Integer, range)
+
+Find `y` in `range` such that `x` ≡ `y` (mod `n`), where `n = length(range)`.
```

Comment:
If we're going to show a mathy expression, we should probably incorporate the offset computation, too.

Reply:
I was trying to keep it short, and copied from `mod(x::Integer, T::Type)`. But indeed the actual code is so short maybe it should just be shown too, have added this.